src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -464,6 +471,110 @@ PRIMARY KEY ({primaryKeys})
             }
         }
 
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            Debug.Assert(!(this._changeMonitor is null));
+
+            return new SqlTriggerMetrics
+            {
+                UnprocessedChangeCount = await this._changeMonitor.GetUnprocessedChangeCountAsync(),
+                Timestamp = DateTime.UtcNow,
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.Cast<SqlTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<SqlTriggerMetrics> context)
+        {
+            return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusCore(int workerCount, SqlTriggerMetrics[] metrics)
+        {
+            const int minMetricSamplesForScaling = 5;
+            const int maxChangesPerWorker = 1000;
+
+            var status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            // Do not make a scale decision unless we have enough samples.
+            if (metrics == null || (metrics.Length < minMetricSamplesForScaling))
+            {
+                return status;
+            }
+
+            // Add worker is count of unprocessed changes per worker exceeds the maximum limit.
+            long lastUnprocessedChangeCount = metrics.Last().UnprocessedChangeCount;
+            this._logger.LogInformation($"Unprocessed change count: {lastUnprocessedChangeCount}, worker count: {workerCount}, maximum changes per worker: {maxChangesPerWorker}.");
+
+            if (lastUnprocessedChangeCount > workerCount * maxChangesPerWorker)
+            {
+                status.Vote = ScaleVote.ScaleOut;
+                this._logger.LogInformation($"Requesting scale-out: Found too many unprocessed changes for table: '{this._userTable.FullName}' relative to the number of workers.");
+                return status;
+            }
+
+            // Check to see if there are no changes occurring in the user table for a while.
+            if (metrics.All(p => p.UnprocessedChangeCount == 0L))
```

Comment:
Are the metrics cleared every once in a while then?

Reply:
Seems to be so. As said in the other comment thread, the queue listener carries similar logic so I assumed that the metrics should be getting cleared, and though of confirming this while testing the published function app. I will update this thread with my observation. BTW, I think I should remove this check as this should be covered entirely by the 'isDecreasing' check below.

Reply:
So, I found the number of metrics to be varying between 10 and 12. The logic for this is entirely dictated by functions host runtime. All metrics populated in last two minutes are considered, which is consistent with my observation. Code for reference:

- [ScaleOptions.cs](https://github.com/Azure/azure-functions-host/blob/379e98010c1d789369df6516b0eb5cc60556cdc7/src/WebJobs.Script/Config/ScaleOptions.cs#L25)
- [TableStorageScaleMetricsRepository.cs](https://github.com/Azure/azure-functions-host/blob/379e98010c1d789369df6516b0eb5cc60556cdc7/src/WebJobs.Script.WebHost/Scale/TableStorageScaleMetricsRepository.cs#L324)